Changes to eligibility framework and member process (#47)
protocolguild/docs: docs/4-roles-expectations.md

#### Qualifications

Note 1: these guidelines will change over time, i.e. become more restrictive in some places and more permissive in others.
Note 2: contributing to the projects/repos referenced below is necessary but does not guarantee Guild eligibility. While this list tries to be explicit by linking to example repos, there are some research areas which can't be linked to a single repo.

Qualifying contributions _must_ be:

- Fully open source, ie. both “source available” and free to fork, modify, redistribute
- The full focus of the individual (anything less receives a partial weighting, see [6.3 weighting](https://protocol-guild.readthedocs.io/en/latest/6-guidelines-for-regular-operation.html#weighting))
- Continuous for at least 6 months ahead of inclusion and ongoing

Qualifying contributions _must_ target at least one of the following projects/areas:

1. Ethereum L1 core protocol maintenance and development

> - Production client implementations: [Erigon](https://github.com/ledgerwatch/erigon), [Geth](https://github.com/ethereum/go-ethereum), [Hyperledger Besu](https://github.com/hyperledger/besu), [Lighthouse](https://github.com/sigp/lighthouse), [Lodestar](https://github.com/ChainSafe/lodestar), [Nethermind](https://github.com/NethermindEth/nethermind), [Nimbus](https://github.com/status-im/nimbus-eth2), [Prysm](https://github.com/prysmaticlabs/prysm), [Teku](https://github.com/ConsenSys/teku)

Reviewer: I don't think a list of clients here is right: why wouldn't silkworm and reth be included?

Reviewer 2: That's a major tangent we can go into: when is a client a peer with the rest of the clients.  Much like the 6 months of time for a contributor to onboard it feels like there should be some "minimum bar," which implies we need standards.  It could be as simple as "has produced a recent mainnet block" but could include other factors like "has developers that attend ACD calls," "unencumbered open source code" (does BSL count?  I'm weakly against including BSL clients), "Interoperates with all/most possible pairs"

Perhaps the simplest way for now is to weasel word it...

```suggestion
- Production client implementations: Currently this includes [Erigon](https://github.com/ledgerwatch/erigon), [Geth](https://github.com/ethereum/go-ethereum), [Hyperledger Besu](https://github.com/hyperledger/besu), [Lighthouse](https://github.com/sigp/lighthouse), [Lodestar](https://github.com/ChainSafe/lodestar), [Nethermind](https://github.com/NethermindEth/nethermind), [Nimbus](https://github.com/status-im/nimbus-eth2), [Prysm](https://github.com/prysmaticlabs/prysm), [Teku](https://github.com/ConsenSys/teku)
```

But I think my bare minimum standard is that any new client has been used to produce mainnet blocks.

Author: @potuz i believe silkworm work is already covered the Erigon umbrella as the contributors work on both. i can reach out to that crew and ask. 

for reth, i think it's still early and the way to make that clear is by specifying the exact requirements - i think @shemnon's suggestions are a great start, but it would be better to have a broader discussion with more members separate from this PR

- Client Testing/security/infra which supports these implementations: [ethereum/tests](https://github.com/ethereum/tests) [ethereum/execution-spec-tests](https://github.com/ethereum/execution-spec-tests)
- Coordination related to upgrades and maintenance: [ethereum/pm](https://github.com/ethereum/pm)

> 2. Research and implementation experiments related to potential protocol changes/refinements

Reviewer 3: Initially I wasn't clear that this was the category under which research falls because I read it as "(research and implementation) experiments ~ research experiments and implementation experiments", and the examples are mostly for the implementation or teams that are in-between. Perhaps it would be more legible to add an example of a research team too?

Author: the old/easy way would be to list RIG 😛 what else could we add?

- fork choice optimizations
- single slot finality
- SSLE
- resource pricing
- mechanism design
- monetary incentives
- PBS

Reviewer 3: Since the new framework appears to be more project-based, yes the list you provided makes sense. I would remove monetary incentives (that sounds like mechanism design, unless you mean monetary policy) and I would add cryptography, maybe group fork choice/ssf under "consensus"? So in a way the three EF research teams (consensus, cryptography, mechanism design)

Author: added these suggestions, thanks!

- [Verkle Tries](https://github.com/gballet/go-verkle) (Verge)
- [Portal Network](https://github.com/ethereum/portal-network-specs) (Purge)

> - EVM Improvements eg. [Ipsilon](https://github.com/ipsilon)

Reviewer 4: do we want an 'eg' here? i'd prefer we list ipsilon, and the reason being EVM Improvements, and others in this same category that we believe qualify can then raise a PR and we discuss, as you describe... 'eg' leaves it less defined, where as raising PR's and explicitly listing the projects added would be clear.

Author: good catch!

3. Spec work resulting from the above (should be implementation agnostic, unopinionated)
- [Execution Specs(EELS)](https://github.com/ethereum/execution-specs)
- [Consensus Specs](https://github.com/ethereum/consensus-specs)

> 4. Protocol Guild itself

Reviewer: This was discussed internally as a very exceptional case and listing it in the core section in the same footing as other projects may raise the idea that we are willing to increase the count of members in this area, something that will meet a strong resistance IMO

Author: putting it all in the same place was just for convenience. happy to move it if others feel the same way

Author: added language and separated out with this commit https://github.com/protocolguild/docs/pull/47/commits/5847f3fcbb029ee60ebb1c16074daae1028f9054

## 4.2 Members

For the Pilot, members of the Protocol Guild will have up to three different functions: [slot holders](https://protocol-guild.readthedocs.io/en/latest/4-roles-expectations.html#slot-holders), [curators](https://protocol-guild.readthedocs.io/en/latest/4-roles-expectations.html#curators) and [signers](https://protocol-guild.readthedocs.io/en/latest/4-roles-expectations.html#signers).

### 4.21 Slot Holders

Slot holders are members of the Guild who have qualified for a placement (slot) in the split contract. Estimated to be around 150-200 individuals, however the actual number may be higher or lower. There is no cap or target for number of slots. Slots can be set to any Ethereum address, including the individuals own, a charity, or another split contract.

#### Qualifications

Note 1: these guidelines will change over time, i.e. become more restrictive in some places and more permissive in others.

> Note 2: contributing to the projects/repos referenced below is necessary but does not guarantee Guild eligibility. While this list tries to be explicit by linking to example repos, there are some research areas which can't be linked to a single repo.

Reviewer 5: ```suggestion

Note 2: contributing to the projects/repos referenced below is necessary but does not guarantee Guild eligibility. While this list tries to be explicit by linking to example repos, there are some research areas which can't be linked to a single repo.
```

Author: maybe i'm reading this wrong - are you deleting note 1 or just adding a new line between both of them?

Qualifying contributions _must_ be:

> - Fully open source, ie. both “source available” and free to fork, modify, redistribute

Reviewer 5: ```suggestion
- Fully open source, i.e. both “source available” and free to fork, modify, redistribute
```

- The full focus of the individual (anything less receives a partial weighting, see [6.3 weighting](https://protocol-guild.readthedocs.io/en/latest/6-guidelines-for-regular-operation.html#weighting))
- Continuous for at least 6 months ahead of inclusion and ongoing

Qualifying contributions _must_ target at least one of the following projects/areas:

1. Ethereum L1 core protocol maintenance and development

- Production client implementations: Currently this includes [Erigon](https://github.com/ledgerwatch/erigon), [Geth](https://github.com/ethereum/go-ethereum), [Hyperledger Besu](https://github.com/hyperledger/besu), [Lighthouse](https://github.com/sigp/lighthouse), [Lodestar](https://github.com/ChainSafe/lodestar), [Nethermind](https://github.com/NethermindEth/nethermind), [Nimbus](https://github.com/status-im/nimbus-eth2), [Prysm](https://github.com/prysmaticlabs/prysm), [Teku](https://github.com/ConsenSys/teku)

> - Client Testing/security/infra which supports these implementations: [ethereum/tests](https://github.com/ethereum/tests) [ethereum/execution-spec-tests](https://github.com/ethereum/execution-spec-tests)

Reviewer 5: ```suggestion
- Client testing/security/infra which supports these implementations: [ethereum/tests](https://github.com/ethereum/tests) [ethereum/execution-spec-tests](https://github.com/ethereum/execution-spec-tests)
```

- Coordination related to upgrades and maintenance: [ethereum/pm](https://github.com/ethereum/pm)

2. Research and implementation experiments related to potential protocol changes/refinements

> - [Verkle Tries](https://github.com/gballet/go-verkle) (Verge)

Reviewer 5: ```suggestion
- [Verkle tries](https://github.com/gballet/go-verkle) (Verge)
```

- [Portal Network](https://github.com/ethereum/portal-network-specs) (Purge)

> - EVM Improvements: [Ipsilon](https://github.com/ipsilon)

Reviewer 5: ```suggestion
- EVM improvements: [Ipsilon](https://github.com/ipsilon)
```

- Consensus work
- Cryptography
- Mechanism design
- Resource pricing

3. Spec work resulting from the above (should be implementation agnostic, unopinionated)

> - [Execution Specs(EELS)](https://github.com/ethereum/execution-specs)

Reviewer 5: ```suggestion
- [Execution specs (EELS)](https://github.com/ethereum/execution-specs)
```

> - [Consensus Specs](https://github.com/ethereum/consensus-specs)

Reviewer 5: ```suggestion
- [Consensus specs](https://github.com/ethereum/consensus-specs)
```

4. Protocol Guild itself
- General comms
- Fundraising
- Research related to the evolution of the Guild itself
- Internal maintenance for the Guild membership

Independent or unaffiliated contributors are considered by the same guidelines as any contributors "officially" part of teams/projects.

> See [6.4 Modifying Projects and Members](URL) for guidelines on adding/removing eligible projects and members.

Reviewer 5: ```suggestion
See [6.4 Modifying Projects and Members](https://protocol-guild.readthedocs.io/en/latest/6-guidelines-for-regular-operation.html#proposing-and-discussing-new-members) for guidelines on adding/removing eligible projects and members.
```

## 4.2 Members

For the Pilot, members of the Protocol Guild will have up to three different functions: [slot holders](https://protocol-guild.readthedocs.io/en/latest/4-roles-expectations.html#slot-holders), [curators](https://protocol-guild.readthedocs.io/en/latest/4-roles-expectations.html#curators) and [signers](https://protocol-guild.readthedocs.io/en/latest/4-roles-expectations.html#signers).

### 4.21 Slot Holders

> Slot holders are members of the Guild who have qualified for a placement (slot) in the split contract. Estimated to be around 150-200 individuals, however the actual number may be higher or lower. There is no cap or target for number of slots. Slots can be set to any Ethereum address, including the individuals own, a charity, or another split contract.

Reviewer 6: ```suggestion
Slot holders are members of the Guild who have qualified for a placement (slot) in the split contract. Estimated to be around 150-200 individuals, however the actual number may be higher or lower. There is no cap or target for number of slots. Slots can be set to any Ethereum address, including the individual's own, a charity, or another split contract.
```
